### PR TITLE
Add problem search result tile

### PR DESCRIPTION
**Pull request by evulins**
Create problem search result tile component with title author, date, and date.

### CODE REVIEW

File: src/components/ProblemIndex/index.tsx [hunk 1]
```diff
@@ -3,26 +3,51 @@ import { FC } from "react";
...
+      />
+      <SearchResultTile
+        text="Kompozycja mazurków."
+        tags={["chopin", "muzyka klasyczna"]}
+        date={new Intl.DateTimeFormat("en-GB", {
+          dateStyle: "medium",
+        }).format(new Date())}
+        author="Fryderyk Chopin"
+        educationStage="Pierwszy etap edukacji"
+      />
```
Comment (bazelajoanna) [inline src/components/ProblemIndex/index.tsx]: Mazurkas <3

**Review by bazelajoanna — APPROVED** [review]
It looks very nice! :)

File: src/components/Button/index.tsx [hunk 2]
```diff
@@ -12,10 +12,11 @@ const Button: FC<ButtonProps> = ({
   disabled = false,
   withDownloadIcon = false,
   children,
+  isHidden = false,
```
Comment (franiakat) [inline src/components/Button/index.tsx:15]: In general once we have props with default values we should put them at the end of the props object. Here we have a specific case, as `children` prop coming from react's `FC` interface is an optional prop - it does not make a difference, but as a good practice we can anyway move children at the top. (This is not something you introduced, but we can improve it now I think ;)) I can elaborate on the call.

File: src/components/ProblemIndex/index.tsx [hunk 3]
```diff
@@ -3,26 +3,50 @@ import { FC } from "react";
...
-        Zaloguj się
-      </Button>
-      <Button isPrimary={false} disabled withDownloadIcon>
-        Zaloguj się
-      </Button>
-    </>
+      <SearchResultTile
+        text="Określenie ciągu. Sposoby opisywania ciągów."
+        tags={["ciągi", "tag"]}
+        date={new Intl.DateTimeFormat("en-GB", {
```
Comment (franiakat) [inline src/components/ProblemIndex/index.tsx]: Actually that could be `pl`, right? ;)

Comment (franiakat) [reply src/components/ProblemIndex/index.tsx]: Usually I'd use `date-fns` for any data related stuff, but this is really simple and static usage, no age counting etc, so it's fine. ;)

File: src/components/ProblemIndex/index.tsx [hunk 4]
```diff
@@ -3,26 +3,50 @@ import { FC } from "react";
...
+        <Button isPrimary={false} disabled>
+          Zaloguj się
+        </Button>
+        <Button isPrimary={false} withDownloadIcon>
+          Zaloguj się
+        </Button>
+        <Button isPrimary={false} disabled withDownloadIcon>
+          Zaloguj się
+        </Button>
+      </Wrapper>
```
Comment (franiakat) [inline src/components/ProblemIndex/index.tsx]: I guess we can remove it now ;)
```suggestion
```

File: src/components/SearchResultTile/index.test.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import { render, screen } from "@testing-library/react";
+
+import SearchResultTile from "./index";
+
+describe("Search Problem Result Tile", () => {
```
Comment (franiakat) [inline src/components/SearchResultTile/index.test.tsx]: Usually I'd go with:
```suggestion
describe("<SearchResultTile />", () => {
```

File: src/components/SearchResultTile/styledComponents.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.1);
+  flex-direction: row;
+  align-items: center;
+  justify-content: space-between;
+  background-color: ${white};
+  margin: 0.3125em 0;
+  padding: 0.5em 1em;
+  border: 0.5px solid ${lightGrey};
+  box-sizing: border-box;
+  border-radius: 5px;
```
Comment (franiakat) [inline src/components/SearchResultTile/styledComponents.tsx:17]: Just a nitpick, you can try to group similar properties together, here border-related, you can have similar groups with fonts and spacings etc.

File: src/components/SearchResultTile/styledComponents.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+const AuthorWrapper = styled(Wrapper)`
+  align-items: center;
+  padding: 0;
+  margin: 0;
+  gap: 0.5em;
+  margin: 1em 0;
+`;
+
+const TagWrapper = styled(Wrapper)`
+  /* align-items: flex-start; */
```
Comment (franiakat) [inline src/components/SearchResultTile/styledComponents.tsx]: leftover?

File: src/components/SearchResultTile/index.test.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+          dateStyle: "medium",
+        }).format(new Date())}
+        author="Maria Salomea Skłodowska-Curie"
+        educationStage="Dowolony etap edukacji"
+      />
+    );
+
+    const date = new Intl.DateTimeFormat("en-GB", {
+      dateStyle: "medium",
+    }).format(new Date());
```
Comment (franiakat) [inline src/components/SearchResultTile/index.test.tsx]: You could reuse it in date props and in an expect, I think? :)

File: src/components/SearchResultTile/index.test.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+    expect(screen.getByText("Określenie ciągu")).toBeInTheDocument();
+    expect(screen.getByText(date)).toBeInTheDocument();
+    expect(
+      screen.getByText("Maria Salomea Skłodowska-Curie")
+    ).toBeInTheDocument();
+    expect(screen.getByText("ciągi")).toBeInTheDocument();
+    expect(screen.getByText("Dowolony etap edukacji")).toBeInTheDocument();
+  });
+  it("shoud displays button", () => {
```
Comment (franiakat) [inline src/components/SearchResultTile/index.test.tsx]:
```suggestion
  it("displays button to see given problem details", () => {
```

File: src/components/SearchResultTile/index.test.tsx [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    }).format(new Date());
+
+    expect(screen.getByText("Określenie ciągu")).toBeInTheDocument();
+    expect(screen.getByText(date)).toBeInTheDocument();
+    expect(
+      screen.getByText("Maria Salomea Skłodowska-Curie")
+    ).toBeInTheDocument();
+    expect(screen.getByText("ciągi")).toBeInTheDocument();
+    expect(screen.getByText("Dowolony etap edukacji")).toBeInTheDocument();
+  });
```
Comment (franiakat) [inline src/components/SearchResultTile/index.test.tsx:30]:
```suggestion
  });
  
```

File: src/components/SearchResultTile/index.test.tsx [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+        tags={["ciągi", "noga"]}
+        date={new Intl.DateTimeFormat("en-GB", {
+          dateStyle: "medium",
+        }).format(new Date())}
+        author="Maria Salomea Skłodowska-Curie"
+        educationStage="Dowolony etap edukacji"
+      />
+    );
+
+    const button = screen.queryByRole("button", { name: "Zobacz treści" });
```
Comment (franiakat) [inline src/components/SearchResultTile/index.test.tsx]: Or
```suggestion
    const button = screen.queryByRole("button", { name: "Zobacz treść" });
```
? : >

File: src/components/SearchResultTile/index.test.tsx [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+    expect(screen.getByText("Zobacz treści")).toBeInTheDocument();
+    expect(button).not.toBeInTheDocument();
```
Comment (franiakat) [inline src/components/SearchResultTile/index.test.tsx:45]: to be or not to be? 😂

File: src/components/SearchResultTile/index.tsx [hunk 13]
```diff
@@ -0,0 +1,46 @@
+import { FC } from "react";
+import Tag from "../Tag";
```
Comment (franiakat) [inline src/components/SearchResultTile/index.tsx]:
```suggestion
import { FC } from "react";

import Tag from "../Tag";
```

File: src/components/SearchResultTile/index.tsx [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+        <AuthorWrapper>
+          <DetailsP>{author}</DetailsP>
+          <DetailsP>•</DetailsP>
+          <DetailsP>{date}</DetailsP>
+        </AuthorWrapper>
+        <TagWrapper>
+          <StageP>{educationStage}</StageP>
+          {tags &&
+            tags.map((tagText: string, index: number) => {
+              return <Tag text={tagText} key={tagText} />;
```
Comment (franiakat) [inline src/components/SearchResultTile/index.tsx]: Or
```suggestion
            tags.map((tagText: string, index: number) => (
              <Tag text={tagText} key={tagText} />;
```

File: src/components/SearchResultTile/styledComponents.tsx [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+
+const StageP = styled(P)`
+  text-transform: uppercase;
+  display: none;
+  color: ${textDetails};
+  font-size: 0.875em;
+  @media ${minDevice.desktopTablet} {
+    display: inline-flex;
+  }
+`;
```
Comment (franiakat) [inline src/components/SearchResultTile/styledComponents.tsx]: I don't think those are actual paragraphs? 🤔  More like spans I think.

**Review by franiakat — CHANGES_REQUESTED** [review]
Niice! I've left a few comments for you. 👩‍🎨

File: src/components/SearchResultTile/index.tsx [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+        <TitleP>{text}</TitleP>
+        <AuthorWrapper>
+          <DetailsP>{author}</DetailsP>
+          <DetailsP>•</DetailsP>
+          <DetailsP>{date}</DetailsP>
+        </AuthorWrapper>
+        <TagWrapper>
+          <StageP>{educationStage}</StageP>
+          {tags &&
+            tags.map((tagText: string, index: number) => {
```
Comment (franiakat) [inline src/components/SearchResultTile/index.tsx]: `index` is not used

File: src/components/SearchResultTile/index.test.tsx (hunk 12 above)
Comment (evulins) [reply src/components/SearchResultTile/index.test.tsx:45]: It is tough question and answer is not simply :D Let's talk about it

File: src/components/SearchResultTile/styledComponents.tsx (hunk 15 above)
Comment (evulins) [reply src/components/SearchResultTile/styledComponents.tsx]: I changed to span and headings as it was on a design.

**Review by franiakat — APPROVED** [review]
sqash

**Pull request merged**